saleor/core/permissions.py

```diff
@@ -0,0 +1,45 @@
+from django.contrib.auth.models import Permission
+from django.shortcuts import get_object_or_404
+
+from saleor.userprofile.models import User
+
+
+MODELS_PERMISSIONS = (('view', 'View Products in Dashboard'),
+                      ('edit', 'Edit Product in Dashboard'))
+
+PERMISSIONS = set([permission[0] for permission in MODELS_PERMISSIONS])
+
+
+def get_user_permissions(user):
+    form_data = {}
+    permissions = Permission.objects.filter(user=user)
+    for permission in permissions:
+        if str(permission.content_type) not in form_data:
```

Comment:
If we used `defaultdict` with `form_data` we wouldn't need this if-else structure.